quora_dataset.py

```diff
@@ -63,78 +62,64 @@ def get_examples(self):
     def get_text_for_qid(self, qid):
         return self._qid2text.get(qid, None)
 
-    def construct_retrieval_task(self, train_size=139306, test_size=12206, retrieval_size=9218):
+    def construct_retrieval_task(self, train_size=139306, test_queries_size=9218, candidates_size=19081):
         # use positive examples to form disjoint sets
-        pos_examples = [e for e in self._examples if e.is_duplicate]
-        pos_qids = set(flatmap([(e.qid1, e.qid2) for e in pos_examples]))
+        pos_qids = set(flatmap([(e.qid1, e.qid2) for e in self._examples]))
         uf = UnionFind(pos_qids)
-        for e in pos_examples:
+        for e in self._examples:
             # each positive example is considered an edge
             uf.union(e.qid1, e.qid2)
 
-        #  disjoint_set is a map of set name to its members
+        # disjoint_set is a map of set name to its members
         disjoint_set = defaultdict(list)
         for qid in pos_qids:
             set_name = uf.find(qid)
             disjoint_set[set_name].append(qid)
 
         # construct training pairs
-        set_names = list(disjoint_set.keys())
-        random.shuffle(set_names)
+        items = list(disjoint_set.items())
+        random.shuffle(items)
+        disjoint_set = dict(items)
```

Comment:
Curious as to why shuffling the key&value pairs and re-creating the dictionary instead of just shuffling the keys?

Reply:
I guess that's because the code below wants to iterate the values of disjoint_set later

Reply:
I wanted to shuffle the entire dict (assuming we're on python 3.7 and it's ordered) to randomize the choice of training and test data. Shuffling the keys without recreating the dict would be just as good if we only had one loop, but the first one removes data from `disjoint_set` and the second one iterates over only what's left. I'm not super attached to this way of processing though, if you'd like to simplify it somehow we absolutely could!